typescript/sdk/src/providers/MultiProvider.ts

```diff
@@ -1,139 +1,272 @@
-import { Signer, ethers } from 'ethers';
-
-import { ChainMap, ChainName, IChainConnection, Remotes } from '../types';
-import { MultiGeneric } from '../utils/MultiGeneric';
-import { objMap, pick } from '../utils/objects';
-
-import { ChainConnection } from './ChainConnection';
-
-export class MultiProvider<
-  Chain extends ChainName = ChainName,
-> extends MultiGeneric<Chain, ChainConnection> {
-  constructor(chainConnectionConfigs: ChainMap<Chain, IChainConnection>) {
-    super(
-      objMap(
-        chainConnectionConfigs,
-        (_, connection) => new ChainConnection(connection),
-      ),
-    );
+import { Debugger, debug } from 'debug';
+import {
+  BigNumber,
+  ContractReceipt,
+  ContractTransaction,
+  PopulatedTransaction,
+  Signer,
+  providers,
+} from 'ethers';
+
+import { types } from '@hyperlane-xyz/utils';
+
+import { ChainMetadata, chainMetadata } from '../consts/chainMetadata';
+import { CoreChainName, TestChains } from '../consts/chains';
+import { ChainMap, ChainName } from '../types';
+import { pick } from '../utils/objects';
+
+type Provider = providers.Provider;
+
+interface Params {
```

Comment:
I feel like the optionality/default-ness of these parameters should, needs to be specified/documented pretty well. Not clear that for example MP will construct the provider from metadata if none is given, but will use the provider if it is.

Reply:
Okay I'll add comments and make sure it's documented